ET use compiler macros in ops

ivayloen opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #13638
* #13637
* #13636

Use new compiler macros to fix operator library builds.
Additional issues,
isnan and signbit supported types are only on float, double and long double. Apply static cast to avoid msvc compiler error.

Differential Revision: [D78854136](https://our.internmc.facebook.com/intern/diff/D78854136/)

ivayloen committed:
ET use compiler macros in ops

Use new compiler macros to fix operator library builds.
Additional issues,
isnan and signbit supported types are only on float, double and long double. Apply static cast to avoid msvc compiler error.

Differential Revision: [D78854136](https://our.internmc.facebook.com/intern/diff/D78854136/)

[ghstack-poisoned]

ivayloen committed:
Update on "ET use compiler macros in ops"


Use new compiler macros to fix operator library builds.
Additional issues,
isnan and signbit supported types are only on float, double and long double. Apply static cast to avoid msvc compiler error.

Differential Revision: [D78854136](https://our.internmc.facebook.com/intern/diff/D78854136/)

[ghstack-poisoned]

GregoryComer commented:
Overall, these changes look reasonable to me. The main concern that I want to address is ensuring that we can have GitHub CI coverage for the platforms we're supporting. As @mergennachin mentioned, if you can use clang-cl internally, that would align with what we're supporting via CMake. We could add CI coverage for the Windows build via buck to guard this.

ivayloen committed:
Update on "ET use compiler macros in ops"


Use new compiler macros to fix operator library builds.
Additional issues,
isnan and signbit supported types are only on float, double and long double. Apply static cast to avoid msvc compiler error.

Differential Revision: [D78854136](https://our.internmc.facebook.com/intern/diff/D78854136/)

[ghstack-poisoned]
